src/ansys/fluent/core/solver/flobject.py

```diff
@@ -1005,8 +1030,26 @@ def get_completer_info(self, prefix="", excluded=None) -> List[List[str]]:
 class Command(Action):
     """Command object."""
 
-    def __call__(self, **kwds):
-        """Call a query with the specified keyword arguments."""
+    def __call__(self, purpose: Optional[str] = None, **kwds):
+        """Call a command with the specified keyword arguments."""
+        file_purpose = None
+        if hasattr(self, "file_name"):
+            file_purpose = self.file_name.get_attr(_InlineConstants.file_purpose)
+        elif hasattr(self, "file_name_list"):
+            file_purpose = self.file_name_list.get_attr(_InlineConstants.file_purpose)
+        elif hasattr(self, "filename"):
+            file_purpose = self.filename.get_attr(_InlineConstants.file_purpose)
+        if "file_name" in kwds:
+            kwds.update(
+                dict(
+                    file_name=os.path.basename(kwds["file_name"])
+                    if self.remote_file_handler.is_configured()
+                    else kwds["file_name"]
+                )
+            )
+        file_purpose = purpose if purpose else file_purpose
+        if file_purpose == "input" and self.remote_file_handler.is_configured():
+            self.remote_file_handler.upload(file_name=kwds["file_name"])
```

Comment:
@hpohekar 
Key point # 1: We shall use the purpose attribute to generate classes, not as a runtime switch. Refer back to #2195 where it is stated
Key point # 2: Irrespective of which argument names are used or not used in the API, we shall not depend on those names in the logic. That was not part of the design, and none of the PyFluent code shall be written on that basis
Key point # 3: `is_configured` shall not be part of the service interface. This has already been pointed out. Such a method is not required because either a valid service is provided or none at all (I made a similar statement in a previous comment).

Reply:
@seanpearsonuk 

Key point # 3: - removed.

Key point # 2: - How to handle following PyPim behavior ?

![image](https://github.com/ansys/pyfluent/assets/106588300/47c4772e-18b5-4c91-8430-0af66c5e6eee)

Reply:
related to Key point # 2:

We can remove following code but then Ansys Lab reads file from absolute path i.e. current working directory only. If we are OK with this behavior then we will remove it.

![image](https://github.com/ansys/pyfluent/assets/106588300/f390d436-d47c-49f4-ae26-fdbc402bdd24)

Reply:
related to Key point # 1:

Ok. We will continue our earlier approach - https://github.com/ansys/pyfluent/pull/2238/files

We will use `file_purpose` attribute instead of `type` to generate classes.

Reply:
Re: "Key point # 2: - How to handle following PyPim behavior ?", the comment is about depending on API argument names like `file_name`, `FileName` etc. It's nothing to do with PyPIM.